sympy/matrices/common.py

```diff
@@ -2065,6 +2065,8 @@ def __pow__(self, num):
                     except (AttributeError, MatrixError):
                         pass
                 return a._eval_pow_by_recursion(num)
+            elif num.is_Number!=True and num.is_negative==None:
+                return ("{}**{}").format(a,num)
```

Comment:
This should not return a string.

Reply:
Okk @asmeurer , so how to return the expression itself?

Reply:
Is it ok in new commit.